Source/render/ShapeRenderer.swift

```diff
@@ -129,6 +129,8 @@ class ShapeRenderer: NodeRenderer {
             ctx!.setFillColor(color.toCG())
         } else if let gradient = fill as? Gradient {
             drawGradient(gradient, ctx: ctx, opacity: opacity)
+        } else if let pattern = fill as? Pattern {
```

Comment:
It is not related to this PR (recommendation only), maybe we should convert this (in another similar place too) if constructions with many cases to appropriate switch cycle?

`switch fill {
 case let gradient = fill as? Gradient:
}`

Reply:
I honestly don't see why switch would be better in this case (I think of it as a thingy for enums only), but I can do it, since I don't care very much